underpants.go

```diff
@@ -111,6 +127,7 @@ func (u *user) encode(key []byte) (string, error) {
   if err := json.NewEncoder(w).Encode(u); err != nil {
     return "", err
   }
+  w.Close()
```

Comment:
I'm not super familiar with how `defer` works -- would executing `Close()` at the end of the function call be possibly closing the encoder too late since `b.String()` and `h.Sum()` have already been called on the un-closed encoder?